x/halo/keeper/keeper.go

```diff
@@ -1,81 +1,101 @@
 package keeper
 
 import (
+	"context"
 	"encoding/json"
 	"fmt"
 
-	sdkerrors "cosmossdk.io/errors"
+	"cosmossdk.io/collections"
+	"cosmossdk.io/core/store"
+	"cosmossdk.io/errors"
+	"cosmossdk.io/math"
+
 	"github.com/cosmos/cosmos-sdk/codec"
 	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
-	storetypes "github.com/cosmos/cosmos-sdk/store/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/noble-assets/halo/x/halo/types"
+	"github.com/noble-assets/halo/x/halo/types/aggregator"
+	"github.com/noble-assets/halo/x/halo/types/entitlements"
 )
 
 type Keeper struct {
-	cdc      codec.Codec
-	storeKey storetypes.StoreKey
-
 	Denom      string
 	Underlying string
 
+	Schema       collections.Schema
+	storeService store.KVStoreService
+
+	Owner  collections.Item[string]
+	Nonces collections.Map[[]byte, uint64]
+
+	AggregatorOwner collections.Item[string]
+	LastRoundId     collections.Sequence
+	NextPrice       collections.Item[math.Int]
+	Rounds          collections.Map[uint64, aggregator.RoundData]
+
+	EntitlementsOwner  collections.Item[string]
+	Paused             collections.Item[bool]
+	PublicCapabilities collections.Map[string, bool]
+	RoleCapabilities   collections.Map[collections.Pair[string, uint64], bool]
+	UserRoles          collections.Map[collections.Pair[[]byte, uint64], bool]
```

Comment:
I think its better to keep the collections not-publicly exposed. Since there are keeper methods available anyway, for other modules to interact with. Wouldn't want other modules to be able to directly modify this module's state.
Not a big thing ofc, but a personal preference.
Feel free to ignore 😅